lib/post_merger.rb

```diff
@@ -38,29 +39,25 @@ def merge
 
   private
 
-  def ensure_at_least_two_posts
-    @posts.count >= 2
-  end
-
   def ensure_same_topic!
-    unless @posts.map(&:topic_id).uniq.length == 1
+    if @posts.map(&:topic_id).uniq.size != 1
       raise CannotMergeError.new(I18n.t("merge_posts.errors.different_topics"))
     end
   end
 
   def ensure_same_user!
-    unless @posts.map(&:user_id).uniq.length == 1
+    if @posts.map(&:user_id).uniq.size != 1
       raise CannotMergeError.new(I18n.t("merge_posts.errors.different_users"))
     end
   end
 
-  def ensure_staff_user!(guardian)
-    raise Discourse::InvalidAccess unless guardian.is_staff?
+  def ensure_can_merge!(guardian)
+    raise Discourse::InvalidAccess unless guardian.can_moderate_topic?(@posts[0].topic)
```

Comment:
This looks like it will allow category group mods to merge posts belonging to topics, if that topic itself belongs to a category that they have moderator permissions for. Is this the intention?

(That sounds like a fine feature to add, but it doesn't match the commit message.)